tests/functional/network_policy/test_network_policy.py

```
@@ -0,0 +1,453 @@
+    def test_policy_structure_validation(self, network_policies_present):
+        """
+        A-4: Validate NetworkPolicies follow ODF design principles.
+        """
+        all_issues = []
+        for policy in network_policies_present:
+            result = verify_policy_structure(policy)
+            if not result["valid"]:
+                all_issues.extend(result["issues"])
+
+        assert not all_issues, (
+            f"Policy structure issues:\n" + "\n".join(all_issues)
```

Ruff (0.16.0)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)